src/core/client_config/lb_policies/pick_first.c

```diff
@@ -171,10 +174,16 @@ int pf_pick(grpc_exec_ctx *exec_ctx, grpc_lb_policy *pol, grpc_pollset *pollset,
             grpc_connected_subchannel **target, grpc_closure *on_complete) {
   pick_first_lb_policy *p = (pick_first_lb_policy *)pol;
   pending_pick *pp;
+  grpc_connected_subchannel *selected = GET_SELECTED(p);
```

Comment:
Can we get some comments here? I would suggest before this block something like
/* Check atomically if there is a subchannel selected for lock-free fast path*/

and then after the block
/* No subchannel selected yet, so acquire lock and then attempt again */